pennylane/grouping/group_observables.py

```diff
@@ -0,0 +1,252 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the high-level Pauli word partitioning functionality used in measurement optimization.
+"""
+
+from pennylane.wires import Wires
+from pennylane.grouping.utils import (
+    convert_observables_to_binary_matrix,
+    binary_to_pauli,
+    are_identical_pauli_words,
+    get_qwc_compliment_adj_matrix,
+)
+from pennylane.grouping.graph_colouring import largest_first, recursive_largest_first
+import numpy as np
+
+GROUPING_TYPES = ["qwc", "commuting", "anticommuting"]
+GRAPH_COLOURING_METHODS = {"lf": largest_first, "rlf": recursive_largest_first}
+
+
+class PauliGroupingStrategy:  # pylint: disable=too-many-instance-attributes
+    """
+    Class for partitioning a list of Pauli words according to some binary symmetric relation.
+
+    Partitions are defined by the binary symmetric relation of interest, e.g. all Pauli words in a
+    partition being mutually commuting. The partition is accomplished by formulating the list of
+    observables as a graph where nodes represent individual Pauli words and connections between
+    nodes corresponds to the two Pauli words satisfying the symmetric binary relation. Obtaining
```

Comment:
I suggest cleaning up the wording here a bit: "The fewest number of partitions covering the list of Pauli words can then be obtained by..."

Reply:
Thanks, I also think 'covering the list of Pauli words' is redundant, since a partition is typically understood to be a decomposition of a set into disjoint subsets, so of course a partition will be a cover.